### ISSUE TITLE
Make gmpy2 optional

### ISSUE THREAD

**Issue by NCGThompson** [cryptovoting/damgard-jurik#4]
gmpy2 relies on having gmp installed, which can be a hassle, especially is someone wants to implement this library in a stand-alone script to be run with Python3. I suggest using PyCryptodome for math functions related to cryptography (e.g. `isPrime()`). PyCryptodome will use either gmpy or gmpy2 if it is installed, but will run without it. For performance of long integer arithmetic, use the same strategy as PyCryptodome: use `mpz` if it is available but fallback on to the built in long type.

**Comment by nickboucher** [comment]
I like this idea! gmpy2 is a bit of a pain to setup, so this could improve usability while retaining better performance for those who want it.

@NCGThompson if you or anyone else make a PR for this, I'd be happy to review it. Otherwise, I'll keep this open as enhancement to implement as time permits.

**Comment by NCGThompson** [comment]
@nickboucher What was the purpose 714cdd1b318797d85d76728f3faaa8b4f7d96056? Before the commit, was the code stable and efficient? I'm thinking I might just cherrypick the change for a specific file out, so no external libraries are used for prime generation, even if they are available.

**Comment by nickboucher** [comment]
I believe the purpose was primarily to replace Python standard lib functions with gmpy2 calls, as well as some minor logging and naming changes. It should have been stable before that, and that commit ought to be a good place to see where gmpy2 dependencies are used if you're only looking to patch a file locally.

**Comment by NCGThompson** [comment]
> Before the commit, was the code stable and efficient? 

No. [The primality check](https://github.com/cryptovoting/damgard-jurik/blob/c64ac7fee4e4709a83a90276e047aabe62f58693/cryptovote/cryptovote/prime_gen.py#L21-L26) was problematic:

```python3
# TODO: how many iterations?
for a in range(2, 6):
    if pow(a, p - 1, p) != 1:
        return False
return True
```

There are not enough iterations, and there are certain types of composite numbers that always pass the test.
